### DISCUSSION TITLE
How to build a higher version of Linux Kernel?

### DISCUSSION THREAD

**Discussion by CunyangWei** [gem5/gem5#1573] [Q&A]
I found in here [linux-configs](https://github.com/gem5/gem5-resources/tree/develop/src/linux-kernel/linux-configs) that the latest version of linux kernel for X86 is 5.4.49. 

But I want to use some new features of linux kernel. I try to build the linux-5.15, and use fullsystem simulation. But GEM5 alway output _panic_
. So, I just want to know is there any thing I need to do when I build a linux kernel?

Besides, I found in this [config](https://github.com/gem5/gem5-resources/blob/develop/src/linux-kernel/linux-configs/config.x86.5.4.49) that it seems do not support NUMA (# CONFIG_NUMA is not set). Why is that? Is this because GEM5 do not support NUMA?

**Comment by powerjg** [comment]
1. gem5 supports NUMA. But, you have to set the correct configuration parameters. E.g., on x86 you have to set the E820 entries and the MDAT correctly. We have no examples of this.
2. You can use `make oldconfig` to build a new kernel. New kernels should work with gem5. I have tested up to 6.9.9. The panic is likely unrelated to the kernel. You need to include details of your error message for the community to be able to help.
